lib/charms/opensearch/v0/opensearch_plugins.py

```diff
@@ -458,6 +470,9 @@ def config(self) -> OpenSearchPluginConfig:
             secret_entries_to_del = [...]|{...},
         )
         """
+        if not self._extra_config.get("access-key") or not self._extra_config.get("secret-key"):
+            logger.error("Missing AWS access-key and secret-key configuration")
+            return OpenSearchPluginConfig()
```

Comment:
What will happen in this case? where are those missing pre-confitions handled?

Reply:
Hi @Mehdi-Bendriss, handled [here, in opensearch_backups lib](https://github.com/canonical/opensearch-operator/blob/b7187bfeda061a641d8c8480a5b3836db38ea61b/lib/charms/opensearch/v0/opensearch_backups.py#L658). This lib has the view of the entire charm, opensearch, etc; it can decide if this is critical enough or not and set the juju status. The plugin, on the other hand, can only inform the plugin manager "I am not ready to be configured / disabled yet" and that can happen every time the plugin_manager hit `run()`, as it will try to check each plugin in its list.

Reply:
@Mehdi-Bendriss it should actually raise an error. I fixed that